### PR TITLE
Depends.py - add exclusive domain tests to key exchange testing

### LINKED ISSUE #6439: Depends.py enhancement: key exchange tests
After merging https://github.com/Mbed-TLS/mbedtls/pull/6385, there are configurations using various key exchange types, but there are no performed tests.
This task is to change the current solution:
```
  'kex': ExclusiveDomain(key_exchange_symbols,
                         [build_command + ['lib'],
                          build_command + ['-C', 'programs']]),
```
to
```
  'kex': ExclusiveDomain(key_exchange_symbols, build_and_test),
```
And fix all issues that arise.

### PR DESCRIPTION
**Pull request by AndrzejKurek**
Introduce exclusive domain testing to kex domain in depends.py.
Fixed dependencies for some tests using the test RSA certificate and key from `certs.c` that require the usage of specific ciphersuites.

Fixes #6439.

## Gatekeeper checklist

- [x] **changelog** not required - test only
- [x] **[backport added](https://github.com/Mbed-TLS/mbedtls/pull/6944)**
- [x] **tests** provided - that's what the PR is about

### CODE REVIEW

**Review by mpg — APPROVED** [review]
LGTM

**Review by valeriosetti — APPROVED** [review]
LGTM

**Comment by mpg** [comment]
Fully approved and passing CI, now only needs a backport. @AndrzejKurek

**Comment by AndrzejKurek** [comment]
Backport addded [here](https://github.com/Mbed-TLS/mbedtls/pull/6944).

**Comment by AndrzejKurek** [comment]
Forward-ported some of the changes picked up in the backport.

**Review by mpg — APPROVED** [review]
LGTM

File: tests/scripts/depends.py [hunk 1]
```diff
@@ -421,9 +421,7 @@ def __init__(self, options):
                                           '|MBEDTLS_SHA384_'),
             # Key exchange types. Only build the library and the sample
             # programs.
-            'kex': ExclusiveDomain(key_exchange_symbols,
-                                   [build_command + ['lib'],
-                                    build_command + ['-C', 'programs']]),
+            'kex': ExclusiveDomain(key_exchange_symbols, build_and_test),
```
Comment (mpg) [inline tests/scripts/depends.py:424]: Sorry for not noticing before, but the comment above is now outdated and should be removed.

**Review by mpg — CHANGES_REQUESTED** [review]
Missed something previously, sorry. (Same for the backport.)

File: tests/scripts/depends.py (hunk 1 above)
Comment (AndrzejKurek) [reply tests/scripts/depends.py:423]: Wow, how did I miss that... Fixed, thanks!

**Review by mpg — APPROVED** [review]
LGTM

**Review by valeriosetti — APPROVED** [review]
LGTM! :)

**Pull request merged**